### PR TITLE
bitnami/dataplatform-bp1 addition of observability framework

### PR DESCRIPTION
**Pull request by gkulkarni32**
**Description of the change**

Addition of wavefront observability framework for the dataplatform
Addition of the values.schema.json for the form layout in kubeapps

**Benefits**

we will be able to provide observability framework to the users
we will be able to provide a form layout in kubeapps portal 

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/dataplatform-bp1/values.yaml
Comment (marcosbc) [inline bitnami/dataplatform-bp1/values.yaml:81]: Not sure if this is the best / more consistent way for enabling metrics, I'm not a big fan of it. @javsalgar what do you think?

File: bitnami/dataplatform-bp1/values.yaml
Comment (marcosbc) [inline bitnami/dataplatform-bp1/values.yaml:249]: Why are we adding these options? Not sure if this is something that should be enabled by default. @javsalgar what do you think?

File: bitnami/dataplatform-bp1/values.yaml
Comment (marcosbc) [inline bitnami/dataplatform-bp1/values.yaml:288]: Could you fix the indentation for this comment?

```suggestion
    ## Rules based discovery configuration
    ## Ref: https://github.com/wavefrontHQ/wavefront-kubernetes-collector/blob/master/docs/discovery.md
    ##
```

**Review by marcosbc — COMMENTED** [review]
Thanks for the contribution! It mostly looks good. However, I'm requesting @javsalgar to check a couple of things and also to have another check of this PR in case I missed anything.

File: bitnami/dataplatform-bp1/values.yaml
Comment (javsalgar) [reply bitnami/dataplatform-bp1/values.yaml:81]: Actually, is this template rendered in the proper subchart?

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [inline bitnami/dataplatform-bp1/Chart.yaml:24]: It is needed to specify here the major branch for the subcharts
```suggestion
  - condition: wavefront.enabled
    name: wavefront
    repository: https://charts.bitnami.com/bitnami
    version: 1.x.x
```

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [inline bitnami/dataplatform-bp1/Chart.yaml:33]: It is also needed to add the icon to be rendered in the different hubs and/or marketplaces:
```suggestion
home: https://github.com/bitnami/charts/tree/master/bitnami/dataplatform-bp1
icon: https://bitnami.com/assets/stacks/dataplatform-bp1/img/dataplatform-bp1-stack-220x234.png
keywords:
```

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [inline bitnami/dataplatform-bp1/Chart.yaml:59]: Is backward compatibility guaranteed? If the changes in this PR are not going to break the upgrades from the previous version, we only need to bump the chart version in a minor instead of a major
```suggestion
version: 0.2.0
```

File: bitnami/dataplatform-bp1/values.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/values.yaml:249]: @marcosbc This is basically a solution chart to deploy kafka/spark/solr together. The intention is to provide the pod affinity rules + resource optimization to provide recommended kubernetes node count.

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (miguelaeh) [inline bitnami/dataplatform-bp1/Chart.yaml:40]: Typo here

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (miguelaeh) [inline bitnami/dataplatform-bp1/Chart.yaml:32]: I think we can already add the icon link pointing to the bitnami one.

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [inline bitnami/dataplatform-bp1/Chart.yaml:4]: As there is no standalone application behind this chart, let's use the same app version as the one used as the chart version (see below comment)
```suggestion
appVersion: 0.2.0
```

File: bitnami/dataplatform-bp1/values.schema.json
Comment (miguelaeh) [inline bitnami/dataplatform-bp1/values.schema.json:1]: I think this should not be necessary. Are you adding it because of certain reason?

File: bitnami/dataplatform-bp1/README.md
Comment (carrodher) [inline bitnami/dataplatform-bp1/README.md:171]: The wavefront subcharts parameters should be added to this table

File: bitnami/dataplatform-bp1/values.yaml
Comment (miguelaeh) [reply bitnami/dataplatform-bp1/values.yaml:81]: No, it would not be rendered in the subchart. I think we should use here just a boolean value here.

File: bitnami/dataplatform-bp1/values.yaml
Comment (miguelaeh) [reply bitnami/dataplatform-bp1/values.yaml:249]: AFAIK this is how the OCTO team wants the Helm Chart to be deployed by default. You can also see other affinity rules set in the rest of components

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you for the PR!
Please check the comments

**Review by carrodher — COMMENTED** [review]
Thanks for the contribution, I just added some comments mainly related to some generic files like Chart.yaml, deps and README

File: bitnami/dataplatform-bp1/values.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/values.yaml:249]: yes thats right @miguelaeh

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/Chart.yaml:32]: can you please give me the link, will update

File: bitnami/dataplatform-bp1/values.schema.json
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/values.schema.json:1]: yes, we would like to have a form layout as well in kubeapps for user experience.

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/Chart.yaml:4]: In the first release, the appVersion was 1.0.0 and the chart version was 0.1.0. 
According to us, this is a major release for the data platform chart, hence we went with 2.0.0 and 1.0.0.

File: bitnami/dataplatform-bp1/README.md
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/README.md:171]: done

File: bitnami/dataplatform-bp1/values.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/values.yaml]: I changed this, please check.

File: bitnami/dataplatform-bp1/values.yaml
Comment (marcosbc) [reply bitnami/dataplatform-bp1/values.yaml:249]: Fine then

File: bitnami/dataplatform-bp1/values.yaml
Comment (marcosbc) [reply bitnami/dataplatform-bp1/values.yaml]: @gkulkarni32 Could you change this to use a plain boolean then?

**Comment by marcosbc** [comment]
Added a request to remove the templating from `metrics.enabled` keys, as it would not be rendered properly in the subchart.

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [reply bitnami/dataplatform-bp1/Chart.yaml:4]: The `appVersion` is not used by Helm to generate the package tarball, it is considered as metadata. This version points to the version of the bundled application in the Helm Chart. For example, in the PostgreSQL Helm chart, this field is `appVersion: 11.11.0` which is the version of the PostgreSQL container used by the Helm Chart.

Then we have the `version` field which is the version of the chart itself (nothing related to the application inside the chart). This version should be modified every time the chart is modified (changes in the values.yaml, notes.txt, k8s manifests, etc). This version follows [semver](https://semver.org/spec/v2.0.0.html). According to semver, the patch version is bumped when the changes are fixing bugs, the minor version when some feature is added and the major version when the backward compatibility is not guaranteed. +info: https://helm.sh/docs/topics/charts/#charts-and-versioning

---------------------------------------

The above explanation is how the version works in Helm as a general rule. In the case of this dataplatform chart, there are some exceptions because there is not any main component bundled in the chart to be used to set the `appVersion` according to it, we can say this chart is not bundling an application so we can consider the chart itself as the product/application, that is why we are suggesting to use the same version in the `appVersion` and `version` field.

In the rest of the charts, the `appVersion` is set according to the upstream maintainers, I mean, if PostgreSQL releases a new version such as 12.4.5 and we update the chart to use it, the `appVersion` is going to reflect this change.

The `version` is not metadata, this field is used to package the chart and it is important to follow the semver directives

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [reply bitnami/dataplatform-bp1/Chart.yaml:61]: Per the above explanation, this field should be updated using semver, in this case as it is a feature addition and not a backward incompatibility we should bump the `version` in a minor.

Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/Chart.yaml]: Updated the version as per the comments. Chart version now is 0.2.0 and appVersion is 1.0.0.

**Review by marcosbc — APPROVED** [review]
LGTM! I would also like to get a review from another team member before merging.

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [reply bitnami/dataplatform-bp1/Chart.yaml]: @miguelaeh maybe you have more context about this topic and I'm missing something, what do you think the relationship between `appVersion` and `version` should be?

Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/Chart.yaml]: I have made the changes to the versions, please have a look.

Comment (carrodher) [reply bitnami/dataplatform-bp1/Chart.yaml]: Thanks for the change, now the `version` was bumped in a minor what is fine, but the `appVersion` field is still pointing to `1.0.0` that was the original value (probably wrong in the first iteration of the Helm chart).

The topic is: `version` is going to be updated at the moment in which there is any change in the chart (even a minor change in a comment), but what about `appVersion`? The purpose of this field is to serve as metadata to show what is the version used by the bundled application, but in this case, there is not a single application. What is your plan to bump `appVersion`? Where should we look to detect that there is a new version? What we usually do in Bitnami in this kind of chart where there is not a single app is set the same value used in `version` and bump both values at the same time.

Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/Chart.yaml]: Hi @carrodher Considering we had 1.0.0 as the appversion in the first one, we can let that be. 
The idea is to have different sizes of the data platform clusters.
Example: Current appversion 1.0.0 is related to 'Small' data platform cluster. Next major version i.e. appversion 2.0.0 will be when we add the 'Medium'/'Large' data platform cluster. 
Hope this clarifies.

File: bitnami/dataplatform-bp1/README.md
Comment (miguelaeh) [inline bitnami/dataplatform-bp1/README.md:44]:
```suggestion
The data platform can be optionally deployed with the wavefront observability framework. In that case, the wavefront collectors will be set up as a DaemonSet to collect the Kubernetes cluster metrics to enable runtime feed into the Tanzu Observability service. It will also be pre-configured to scrape the metrics from the Prometheus endpoint that each application (Kafka/Spark/Solr) emits the metrics to.
```

I would edit it a bit on this way

File: bitnami/dataplatform-bp1/values.yaml
Comment (miguelaeh) [inline bitnami/dataplatform-bp1/values.yaml:190]: I think this will not be templatized in the subchart so we should use the port value.

File: bitnami/dataplatform-bp1/values.yaml
Comment (miguelaeh) [inline bitnami/dataplatform-bp1/values.yaml:196]: Same here

**Review by miguelaeh — CHANGES_REQUESTED** [review]

File: bitnami/dataplatform-bp1/values.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/values.yaml]: Thanks for pointing this out. I updated the port to `{{ .values.spark.master.webPort}}`

File: bitnami/dataplatform-bp1/values.yaml
Comment (gkulkarni32) [reply bitnami/dataplatform-bp1/values.yaml]: Thanks for pointing this out. I updated the port to `{{ .values.spark.worker.webPort}}`

**Review by miguelaeh — APPROVED** [review]
LGTM!

File: bitnami/dataplatform-bp1/Chart.yaml
Comment (carrodher) [reply bitnami/dataplatform-bp1/Chart.yaml]: Ok, as it is something that you are going to handle due to the lack of application when the chart evolves

**Review by carrodher — APPROVED** [review]
Last request on my side, can you run `helm dep update` and add the _Chart.lock_ file to the PR? Once done, we will be able to merge the PR

**Comment by gkulkarni32** [comment]
Updated Chart.lock file, Please check from your side.

**Pull request merged**
